odl/operator/default_ops.py

```diff
@@ -27,6 +30,45 @@
            'ComplexModulus', 'ComplexModulusSquared')
 
 
+def _scale_op(operator, scalar):
+    """Scale an operator, optimizing for ``scalar=0`` and ``scalar=1``."""
+    if scalar == 0:
+        return ZeroOperator(operator.domain, operator.range)
+    elif scalar == 1:
+        return operator
+    else:
+        return scalar * operator
+
+
+def _lico_ops(a, op1, b, op2):
+    """Linear combination of operators, optimizing trivial cases."""
```

Comment:
Not really I guess. It just struck me that we have this inefficiency in operator arithmetic while writing examples for the `repr` implementation. So I just put it out there to see if we want such a thing.

I'll try to pull it out into a separate PR.

Reply:
The more I think of it, the more I come to the conclusion that it might be a bad idea to do this in the first place. The use case in this PR is rather limited (only `ComplexEmbedding` and friends basically). But the scaling thing could be applied to operators in general, like this
```python
>>> I = odl.IdentityOperator(odl.rn(2))
>>> 2 * I
OperatorLeftScalarMult(IdentityOperator(rn(2)), 2)
>>> 1 * I
IdentityOperator(rn(2))
>>> 0 * I
ZeroOperator(rn(2))
```
But does it really make sense to special-case operator scaling? Another thing to note is that putting it into a function that uses `ZeroOperator` basically breaks subclassing of `Operator`, leading to potentially funny behavior.

I'll leave it for now and just remove the code.